bundles/io/org.eclipse.smarthome.io.transport.mdns/src/main/java/org/eclipse/smarthome/io/transport/mdns/internal/MDNSClientImpl.java

```diff
@@ -276,7 +276,7 @@ private void createJmDNSByAddress(InetAddress address) {
     @Override
     public void onChanged(List<CidrAddress> added, List<CidrAddress> removed) {
         logger.debug("ip address change: added {}, removed {}", added, removed);
-        deactivate();
```

Comment:
And now clear `activeServices´ in deactivate and we should be all set.

Reply:
Done (don't trust the garbage collector ;-).

Reply:
:-)
And you never know whether OSGi keeps the instance alive even if deactivated.